curvefs/src/mds/fs_manager.cpp

```diff
@@ -101,29 +101,25 @@ FSStatusCode FsManager::CreateFs(const std::string& fsName,
 
     // create partition
     auto ret = FSStatusCode::OK;
-    curvefs::mds::topology::CreatePartitionRequest request;
-    curvefs::mds::topology::CreatePartitionResponse response;
-    request.set_fsid(wrapper.GetFsId());
-    request.set_count(1);
-    topoManager_->CreatePartition(&request, &response);
-    if (TopoStatusCode::TOPO_OK != response.statuscode() ||
-        response.partitioninfolist_size() < 1) {
+    PartitionInfo partition;
+    TopoStatusCode topoRet = topoManager_->CreatePartition(wrapper.GetFsId(),
+                                                           &partition);
+    if (TopoStatusCode::TOPO_OK != topoRet) {
         LOG(ERROR) << "CreateFs fail, create partition fail"
-                   << ", request = " << request.ShortDebugString()
-                   << ", error code = " << response.statuscode();
+                   << ", fsId = " << wrapper.GetFsId();
         ret = FSStatusCode::CREATE_PARTITION_ERROR;
     } else {
-        PartitionInfo partition = response.partitioninfolist(0);
-        // get leader
+        // get copyset members
         std::set<std::string> addrs;
-        std::string leader;
         if (TopoStatusCode::TOPO_OK != topoManager_->GetCopysetMembers(
                 partition.poolid(), partition.copysetid(), &addrs)) {
             LOG(ERROR) << "CreateFs fail, get copyset members fail,"
                        << " poolId = " << partition.poolid()
                        << ", copysetId = " << partition.copysetid();
             ret = FSStatusCode::UNKNOWN_ERROR;
         } else {
+            // get leader
+            std::string leader;
```

Comment:
the leader can be changed, I think you should not let the caller know about who is the leader, just pass all replicas' addresses, and do get leader procedure in metaserverclient.

Reply:
> the leader can be changed, I think you should not let the caller know about who is the leader, just pass all replicas' addresses, and do get leader procedure in metaserverclient.

fixed